ignore power-on reset value (85.0 °C)

susisstrolch opened the pull request.

Commit:
ignore power-on reset value (85.0 °C)

TD-er commented on src/_P004_Dallas.ino:
```diff
@@ -154,7 +154,8 @@ boolean Plugin_004(byte function, struct EventStruct * event, String& string)
                 if (Plugin_004_DS_readTemp(addr, &value))
                 {
-                    UserVar[event->BaseVarIndex] = value;
+                    if (value != 85.0)    // ignore power on reset value
```
Do not compare a float like that. You cannot compare floats using `==` or `!=`.
Better to check for errors in the read function `Plugin_004_DS_readTemp` itself and return false when an error occurred.

susisstrolch commented:
It‘s no error - it is a valid returnvalue with propper crc. 
Will have a look about the !=....

TD-er commented:
@susisstrolch Not only the "!=" but better also check it inside the function itself, where the sensor value is still an integer value and perhaps also the error bits are known.

Commit:
DS_readTemp: handle power-on reset value

return false if DS_readTemp is power-on reset value
(0xaa for DS1820|DS18B20, 0x550 for DS18B20|DS1825|DS1822)

susisstrolch commented:
as requested - fix in DS_readTemp...
